File: extensions/offlineAds/src/main/java/com/mxplay/offlineads/exo/oma/AdErrorEvent.java

package com.mxplay.offlineads.exo.oma;

public class AdErrorEvent {

    private final AdError adError;
    private final Object userRequestContext;

    public AdErrorEvent(AdError adError, Object userRequestContext) {
        this.adError = adError;
        this.userRequestContext = userRequestContext;
    }

    public final AdError getError() {
        return this.adError;
    }

    public final Object getUserRequestContext() {
        return this.userRequestContext;
    }


    public interface AdErrorListener {
        void onAdError(AdErrorEvent var1);
    }
}

File: extensions/mxInteractiveMediaAds/src/main/java/com/mxplay/interactivemedia/api/AdErrorEvent.kt
package com.mxplay.interactivemedia.api

class AdErrorEvent {
    val error: AdError
    var ad: Ad? = null
    var userRequestContext: Any? = null
        private set

    constructor(adError: AdError,  ad: Ad? = null) {
        error = adError
        this.ad = ad
    }

    constructor(adError: AdError, userRequestContext: Any?) {
        error = adError
        this.userRequestContext = userRequestContext
    }

    interface AdErrorListener {
        fun onAdError(adErrorEvent: AdErrorEvent)
    }

    interface AdErrorVerificationListener {
        fun onAdErrorVerification(adErrorEvent: AdErrorEvent)
    }
}